app/controllers/recipients_controller.rb

```diff
@@ -0,0 +1,41 @@
+class RecipientsController < ApplicationController
+  
+  before_action :set_message
+
+  # GET /recipients/1
+  def show
+    if @message.authenticated?
+      @message.delete()
+    else
+      redirect_to action: 'authenticate'
+    end
+  end
+
+  # GET /recipients/1/authenticate
+  def authenticate
+  end
+
+  # POST /recipients/1
+  def authenticatedShow
```

Comment:
you can probably do away with this unconventional route, and move everything to the show method.  If you retrieve the message before destroying it, it will still be viewable at the time of request, but after refresh will be gone :)